### ISSUE TITLE
not using default values causes stacktrace for integers

### ISSUE THREAD

**Issue by imod** [swagger-api/swagger-core#1568]
please see: https://github.com/springfox/springfox/issues/1100 as a reference

```
 WARN 2015-12-15 07:51:28,834 [tomcat-http--46][] io.swagger.models.parameters.AbstractSerializableParameter - Illegal DefaultValue  for parameter type integer
java.lang.NumberFormatException: For input string: ""
    at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
    at java.lang.Long.parseLong(Long.java:453)
    at java.lang.Long.valueOf(Long.java:540)
    at io.swagger.models.parameters.AbstractSerializableParameter.getDefault(AbstractSerializableParameter.java:161)
    at sun.reflect.GeneratedMethodAccessor107.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.get(BeanPropertyWriter.java:726)
```

**Comment by peterbugiel** [comment]
I'm experiencing the same problem when NOT setting a `defaultValue` in the `@ApiParam` annotation for an `Integer` parameter. In this case an empty string gets set as default and the `AbstractSerializableParameter#getDefault()` method tries to create a `Long` out of "" --> `NumberFormatException`.

```java
public Object getDefault() {
...
    if("integer".equals(this.type)) {
        return Long.valueOf(this.defaultValue);
    }
...
}
```
An additional check for an empty string in the above snippet would solve the problem.

Thanks in advance!.

**Comment by lucasoares** [comment]
I don't know if is the same issue as this one, but I'm having a slight different error:

```
2018-06-29 16:01:04,007 WARN [parameters.AbstractSerializableParameter] Illegal DefaultValue null for parameter type integer
java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Long.parseLong(Long.java:601)
	at java.lang.Long.valueOf(Long.java:803)
	at io.swagger.models.parameters.AbstractSerializableParameter.getExample(AbstractSerializableParameter.java:412)
	at sun.reflect.GeneratedMethodAccessor128.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
```

Looking at AbstractSerializableParameter.java, this issue fixed the default parsing with #1586 but the example parsing need to be fixed too:

```
@JsonProperty("x-example")
    public Object getExample() {
        if (example == null) {
            return null;
        }
        try {
            if (BaseIntegerProperty.TYPE.equals(type)) {
                return Long.valueOf(example);
```

Right?

**Comment by lucasoares** [comment]
Now I saw #2783, sorry for this reply.
